services/src/pro/datasets/external/copernicus_dataspace/ids.rs

```diff
@@ -0,0 +1,429 @@
+use geoengine_datatypes::{
+    dataset::{DataId, DataProviderId, ExternalDataId, LayerId, NamedData},
+    raster::RasterDataType,
+    spatial_reference::{SpatialReference, SpatialReferenceAuthority},
+};
+use std::str::FromStr;
+use strum::EnumIter;
+use strum_macros::EnumString;
+
+use crate::{
+    error::{Error, Result},
+    layers::listing::LayerCollectionId,
+};
+
+#[derive(Debug, Clone)]
+pub enum CopernicusDataspaceLayerCollectionId {
+    Datasets,
+    Sentinel1,
+    Sentinel2(Sentinel2LayerCollectionId),
+}
+
+#[derive(Debug, Clone)]
+pub enum Sentinel2LayerCollectionId {
+    Products,
+    Product {
+        product: Sentinel2Product,
+    },
+    ProductZone {
+        product: Sentinel2Product,
+        zone: UtmZone,
+    },
+}
+
+#[derive(Debug, Clone)]
+pub enum CopernicusDataspaceLayerId {
+    // Sentinel1,
+    Sentinel2(Sentinel2LayerId),
+}
+
+#[derive(Debug, Clone)]
+pub struct Sentinel2LayerId {
+    pub product: Sentinel2Product,
+    pub zone: UtmZone,
+    pub band: Sentinel2Band,
+}
+
+#[derive(Debug, Clone, Copy, EnumString, strum::Display, EnumIter)]
+pub enum Sentinel2Product {
+    L1C,
+    L2A,
+}
+
+// TODO: are these bands for all sentinel 2 products?
+#[derive(Debug, Clone, Copy, EnumString, strum::Display, EnumIter)]
+pub enum Sentinel2Band {
+    B01,
+    B02,
+    B03,
+    B04,
+    B05,
+    B06,
+    B07,
+    B08,
+    B08A,
+    B09,
+    // B10,
+    B11,
+    B12,
+    // TODO: AOT, CLD, etc. but they have multiple resolutions
```

Comment:
for geo engine we need 3 stacked layers for 10m, 20m and 60m. Are there really bands with multiple resolutions?

Reply:
AOT, CLD, SCL, SNW, WVP are contained in both subdataset 2 with 20m resolution and subdataset 3 with 60m resolution. idk what to do about that.

Reply:
Ah now i get it. So i would suggest to add them all with a suffix ore something like that...
currently we only use the SCL and maybe we can add them as SCL@10m and SCL@20m ... SCL is the Scene Classification Layer

Reply:
2135d686f66af0d24d380d9546c0fb5e24959512